.circleci/config.yml

```diff
@@ -1,14 +1,136 @@
-version: 2
-general:
-  branches:
-    only:
-      - master
-      - publishment
-      - release
+version: 2.1
+
+executors:
+  gdal-build-executor:
+    docker:
+      - image: jamesmcclain/gdal-build-environment:4
+    working_directory: /workdir
+
 jobs:
-  build:
-    machine: true
+  build-linux:
+    executor: gdal-build-executor
+    environment:
+      CC: gcc
+      CXX: g++
+      CFLAGS: "-Wall -Wno-sign-compare -Werror -O0 -ggdb3 -DSO_FINI -D_GNU_SOURCE"
+      BOOST_ROOT: "/usr/local/include/boost_1_69_0"
+      JAVA_HOME: "/usr/lib/jvm/java-8-openjdk-amd64"
     steps:
+      - attach_workspace:
+          at: /tmp/workdir/
       - checkout
-      - run: ~/project/.travis/tests.sh
-      - run: ~/project/.circleci/deploy.sh
+      - run:
+          name: Build Linux binaries and run unit tests
+          command: |
+            ln -s /tmp/c41078a1.tif src/experiments/data/c41078a1.tif
+            make -j4 -C src tests
+            make -j4 -C src/experiments/thread pattern oversubscribe
+      - persist_to_workspace:
+          root: src
+          paths:
+            - libgdalwarp_bindings.so
+
+  build-macos:
+    executor: gdal-build-executor
+    environment:
```

Comment:
CircleCI somehow erases / resets container env variables, so I had to set all of them manually